### PR TITLE
79 equals method should be defined for metaattribute class

### PR DESCRIPTION
**Pull request by Aleh-Yanushkevich**
- Implemented `IEquatable<T>` interface for `MetaAttribute` class.
- Added tests for checking Meta Attribute equivalence.

### CODE REVIEW

**Comment by Aleh-Yanushkevich** [comment]
@nvborisenko  please, have a look a failed test. I'm almost sure, that my changes didn't affect it. Seems, that there is a problem with this unit test.

File: test/ReportPortal.Shared.Tests/Execution/Metadata/MetaAttributeFixture.cs [hunk 1]
```diff
@@ -37,5 +37,32 @@ public void ShouldCastToItemAttribute()
...
             ia.Value.Should().Be("b");
         }
+
+        [Fact]
+        public void TwoMetaAttributesWithTheSameKeysAndValuesShouldBeEquals()
+        {
+            var firstAttribute = new MetaAttribute("a", "b");
+            var secondAttriute = new MetaAttribute("a", "b");
+
+            firstAttribute.Should().BeEquivalentTo(secondAttriute);
```
Comment (nvborisenko) [inline test/ReportPortal.Shared.Tests/Execution/Metadata/MetaAttributeFixture.cs:47]: Please double check whether `BeEquivalentTo` method uses `IEquatable` or `Equals`. In my experience this method is smart and can compare any types of objects.

In any case let's explicitly invoke methods under testing.

**Review by nvborisenko — APPROVED** [review]
Approving, but please resolve 1 comment.

File: test/ReportPortal.Shared.Tests/Execution/Metadata/MetaAttributeFixture.cs (hunk 1 above)
Comment (nvborisenko) [reply test/ReportPortal.Shared.Tests/Execution/Metadata/MetaAttributeFixture.cs:47]: Checked it on codecov, looks good

image

**Comment by Aleh-Yanushkevich** [comment]
> Approving, but please resolve 1 comment.

You have already resolved this comment, as I see. Shall we merge it? ot before have to understand issue with failed test?

**Comment by nvborisenko** [comment]
Please merge, lgtm.

**Pull request merged**